### ISSUE TITLE
Relationship to DomainSets.jl

### ISSUE THREAD

**Issue by zsunberg** [JuliaReinforcementLearning/CommonRLSpaces.jl#14]
I recently discovered [DomainSets.jl](https://github.com/JuliaApproximation/DomainSets.jl). It contains some of the sets we need, for example cartesian products, etc. What should this package's relationship to DomainSets be? I see a few possibilities.

1. No relationship - we act like DomainSets doesn't exist (but perhaps learn from it)
2. We maintain compatibility in case someone wants to use things from DomainSets, but also re-implement the functionality that we want here.
3. We use them as a base package for our interface and encourage using their sets in most cases, and only implement a few things that they don't provide (i.e. we would probably get rid of our `product` and use theirs.)
4. We re-export some of their sets like we do with IntervalSets (this seems bad)
5. We archive this package because DomainSets provides everything we need.

**Comment by findmyway** [comment]
I wish I could have known that package earlier :sob: 

# 3 seems to be more appealing to me. We can archive this package and add those that are not provided into CommonRLInterfaces.jl.

**Comment by zsunberg** [comment]
Yes, it is a pretty nice package, but doesn't have some things that we want, like `rand` and `clamp`, so we will have to work out how to integrate with it.
